src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1226,14 +1226,29 @@ public MemoryStream CheckoutMemoryStream()
                 return new MemoryStream();
             }
 
-            return _cachedMemoryStreams.Pop();
+            var stream = _cachedMemoryStreams.Pop();
+            _sizeOfMemoryStreamCache -= stream.Capacity;
+
+            return stream;
         }
 
+        private const long MemoryStreamCacheThreshold = Constants.Size.Megabyte;
+        private const int MemoryStreamCacheMaxCapacityInBytes = 64 * Constants.Size.Megabyte;
+
+        private long _sizeOfMemoryStreamCache;
         public void ReturnMemoryStream(MemoryStream stream)
         {
+            //We don't want to hold big streams in the cache or have too big of a cache
+            if (stream.Capacity > MemoryStreamCacheThreshold || _sizeOfMemoryStreamCache >= MemoryStreamCacheMaxCapacityInBytes)
+            {
+                return;
```

Comment:
I think that we need to dispose the stream in this case?

Reply:
Disposing a memory stream doesn't release any data, and we don't really care about flags like isOpen being closed

Reply:
Ok